### PR TITLE
More updates

### PR DESCRIPTION
**Pull request by cardoe**
- Refactor bits to eventually allow for multiple versions of Rust in the tree
- Drop PACKAGECONFIG for local-rust and add possible support for multiple build architectures.

### CODE REVIEW

**Comment by cardoe** [comment]
/cc @meta-rust/maintainers @posborne 

As an aside do @meta-rust/maintainers want @posborne added to the alias if @posborne wants to be added?

**Comment by jmesmon** [comment]
@cardoe re: adding @posborne : fine by me. Do I need to fiddle with org membership for that to work?

**Comment by cardoe** [comment]
@jmesmon I can't add him to the team without him being part of the org. So I think so.

**Comment by posborne** [comment]
Sure, being part of the org sounds good to me.

File: recipes-devtools/rust/rust.inc [hunk 1]
```diff
@@ -5,28 +5,16 @@ LICENSE = "MIT | Apache-2.0"
 LIC_FILES_CHKSUM ="file://COPYRIGHT;md5=43e1f1fb9c0ee3af66693d8c4fecafa8"
 
 inherit rust
-require rust-snapshot-2016-05-24.inc
 
 SRC_URI = "\
 	https://static.rust-lang.org/dist/rustc-${PV}-src.tar.gz;name=rust \
+    https://static.rust-lang.org/dist/${RUST_SNAPSHOT}.tar.gz;name=rust-snapshot \
```
Comment (posborne) [inline recipes-devtools/rust/rust.inc]: nit: seems to be mix of tabs/spaces with indentation here.

Comment (cardoe) [inline recipes-devtools/rust/rust.inc]: Should be fixed. Thanks.

**Pull request merged**